resilience.ipynb

```diff
@@ -0,0 +1,205 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Resilience against hardware failures\n",
+    "\n",
+    "Scenario:  We have a cluster that partially consists of preemptible ressources.  That is, we'll have to deal with workers suddenly being shut down during computation.\n",
+    "\n",
+    "Relevant docs: <http://distributed.dask.org/en/latest/resilience.html#hardware-failures>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Increase resilience\n",
+    "\n",
+    "Whenever a worker shuts down, the scheduler will increment the suspicousness counter of _all_ tasks that were assigned (not necessarily computing) to the worker in question.  Whenever the suspiciousness of a task exceeds a certain threshold (3 bu default), the task will be considered broken.  We want to compute many tasks on only a few workers with workers shutting down randomly.  So we expect the suspiciousness of all tasks to grow rapidly.  Let's effectively disable the threshold by setting it to a large number:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask\n",
+    "\n",
+    "# needs to be done before importing anything from dask.distributed \n",
```

Comment:
Is this true? Or does it just need to be done before creating a client / scheduler / workers?

Reply:
Yes.
<https://github.com/dask/distributed/blob/e846991d93054a29e28528224c63db69972ffc9c/distributed/scheduler.py#L76> sets the var `ALLOWED_FAILURES` at which will be used during initialization of the scheduler here <https://github.com/dask/distributed/blob/e846991d93054a29e28528224c63db69972ffc9c/distributed/scheduler.py#L832>.

Reply:
I'm not really sure why this is done. Probably due to a policy of generally avoiding repeated `dask.config.get` calls for performance reasons?

Reply:
It's probably unintentional. Schedulers aren't created that often. https://github.com/dask/distributed/pull/2761